extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/models/ModelPrincipalAuthenticationData.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.extension.persistence.relational.jdbc.models;
+
+import org.apache.polaris.core.entity.PolarisPrincipalSecrets;
+
+public class ModelPrincipalAuthenticationData {
+  // the id of the principal
+  private long principalId;
+
+  // the client id for that principal
+  private String principalClientId;
+
+  // Hash of mainSecret
+  private String mainSecretHash;
+
+  // Hash of secondarySecret
+  private String secondarySecretHash;
+
+  private String secretSalt;
+
+  public long getPrincipalId() {
+    return principalId;
+  }
+
+  public String getPrincipalClientId() {
+    return principalClientId;
+  }
+
+  public String getSecretSalt() {
+    return secretSalt;
+  }
+
+  public String getMainSecretHash() {
+    return mainSecretHash;
+  }
+
+  public String getSecondarySecretHash() {
+    return secondarySecretHash;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
```

Comment:
Why not use `@PolarisImmutable` (on the outer class)?

Reply:
I didn't knew about this module, hence missed it, I just did it the way JPA models does it, may be I can refactor both together in a follow-up ?

Reply:
This relates a bit to my comment on using reflection to get fields from this class (vs. explicit code to map fields to SQL columns). 

If we go with `@PolarisImmutable` we release control over field name, so more weight would be on having explicit mapping code.

I'm fine doing this refactoring later.